lib/scheduler.js

```diff
@@ -71,6 +73,45 @@ exports.scheduler = {
         }, destroyTime);
     },
     /**
+     * Schedules the next notification based on our current step.
+     */
+    scheduleNextNotification: function() {
+        let step = storageManager.get('step');
+
+        // clear any currently scheduled notification timer
+        module.exports.scheduler.clearNotificationTimer();
+        // start autoCloseTimer
+        module.exports.scheduler.autoCloseTimer(intervals.afterInteractionCloseTime);
```

Comment:
Yeah, I am thinking that there should probably one "wrapper" function in `scheduler` in this case that can be called here instead of calling all of the individual functions like this. I am definitely going to make a note of this and get back to it as part of my continued work to improve the code base. Thanks!